lib/ui/profitloss/RecordAmountHeader.dart

```diff
@@ -0,0 +1,207 @@
+import 'package:farmsmart_flutter/utils/RegExInputFormatter.dart';
+import 'package:farmsmart_flutter/ui/profitloss/RecordAmount.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:intl/intl.dart';
+
+class _Constants {
+  static final amountValidator = RegExInputFormatter.withRegex(
+      '^\$|^(0|([1-9][0-9]{0,3}))(\\.[0-9]{0,2})?\$');
+}
+
+class _Strings {
+  static final HINT = "00";
+  static final EMPTY_STRING = "";
+}
+
+class RecordAmountHeaderViewModel {
+  String onAmountChanged;
+  bool isEditable;
+
+  RecordAmountHeaderViewModel({
+    this.onAmountChanged,
+    this.isEditable,
+  });
+}
+
+class RecordAmountHeaderStyle {
+  final TextStyle hintTextStyle;
+  final TextStyle titleTextStyle;
+
+  final EdgeInsets edgePadding;
+
+  final double height;
+  final int maxLines;
+
+  const RecordAmountHeaderStyle({
+    this.hintTextStyle,
+    this.titleTextStyle,
+    this.edgePadding,
+    this.height,
+    this.maxLines,
+  });
+
+  RecordAmountHeaderStyle copyWith({
+    TextStyle hintTextStyle,
+    TextStyle titleTextStyle,
+    EdgeInsets edgePadding,
+    double height,
+    int maxLines,
+  }) {
+    return RecordAmountHeaderStyle(
+      hintTextStyle: hintTextStyle ?? this.hintTextStyle,
+      titleTextStyle: titleTextStyle ?? this.titleTextStyle,
+      edgePadding: edgePadding ?? this.edgePadding,
+      height: height ?? this.height,
+      maxLines: maxLines ?? this.maxLines,
+    );
+  }
+}
+
+class _DefaultStyle extends RecordAmountHeaderStyle {
+  final TextStyle hintTextStyle = const TextStyle(
+    fontSize: 72,
+    fontWeight: FontWeight.w500,
+    color: Color(0x4cff8d4f),
+    letterSpacing: 4.32,
+  );
+
+  final TextStyle titleTextStyle = const TextStyle(
+    fontSize: 72,
+    fontWeight: FontWeight.w500,
+    color: Color(0xFFff8d4f),
+    letterSpacing: 4.32,
+  );
+
+  final EdgeInsets edgePadding = const EdgeInsets.symmetric(horizontal: 32);
+  final double height = 92;
+  final int maxLines = 1;
+
+  const _DefaultStyle({
+    TextStyle hintTextStyle,
+    TextStyle titleTextStyle,
+    EdgeInsets edgePadding,
+    double height,
+    int maxLines,
+  });
+}
+
+const RecordAmountHeaderStyle _defaultStyle = const _DefaultStyle();
+
+class RecordAmountHeader extends StatefulWidget {
```

Comment:
not sure I understand why this is split out into own widget and has its own viewmodel and state.

Reply:
Should I join it all in one file for this PR?